web/docs/tql2/operators/metrics.md

```diff
@@ -275,7 +283,29 @@ The records `input` and `output` have the following schema:
 | `unit`         | `string` | The type of the elements, which is `void`, `bytes` or `events`. |
 | `elements`     | `uint64` | Number of elements that were seen during the collection period. |
 | `approx_bytes` | `uint64` | An approximation for the number of bytes transmitted.           |
-| `batches` | `uint64` | The number of batches included in this metric.                       |
+| `batches`      | `uint64` | The number of batches included in this metric.                  |
+
+### `tenzir.metrics.pipeline`
+
+Contains measurements of data flowing through pipelines, emitted once every 10
+seconds.
+
+| Field         | Type     | Description                                     |
+| :------------ | :------- | :---------------------------------------------- |
+| `timestamp`   | `time`   | The time at which this metric was recorded.     |
+| `pipeline_id` | `string` | The ID of the pipeline these metrics represent. |
+| `ingress`     | `record` | Measurement of data entering the pipeline.      |
+| `egress`      | `record` | Measurement of data exiting the pipeline.       |
+
+The records `ingress` and `egress` have the following schema:
+
+| Field      | Type       | Description                                              |
+| :--------- | :--------- | :------------------------------------------------------- |
+| `duration` | `duration` | The timespan over which this data was collected.         |
+| `events`   | `uint64`   | Number of events that passed through during this period. |
+| `bytes`    | `uint64`   | Approximate number of bytes that passed through.         |
+| `batches`  | `uint64`   | Number of batches that passed through.                   |
+| `internal` | `bool`     | True if the data flow is considered internal to Tenzir.  |
```

Comment:
I think a cumulative counter would be highly beneficial as well to quickly work with the total.

Reply:
That's impossible to add to the metric without retrieving older metrics, as the pipeline (or even the entire node) may have restarted in the middle.

Reply:
So then I need to store the historical metrics if I want this. Fine.